passes/memory/memory_share.cc

```diff
@@ -665,9 +665,18 @@ struct MemoryShareWorker
 	// Setup and run
 	// -------------
 
-	MemoryShareWorker(RTLIL::Design *design, RTLIL::Module *module) :
-			design(design), module(module), sigmap(module)
+	MemoryShareWorker(RTLIL::Design *design) :
+			design(design), modwalker(design)
 	{
+	}
+
+	void operator()(RTLIL::Module* module)
+	{
+		this->module = module;
+		sigmap.set(module);
+		sig_to_mux.clear();
+		conditions_logic_cache.clear();
+
 		std::map<std::string, std::pair<std::vector<RTLIL::Cell*>, std::vector<RTLIL::Cell*>>> memindex;
```

Comment:
Kind of ugly.  Why not move the `memindex` declaration to the very top of the function?